src/main/webapp/src/app/find-gardens-page/find-gardens.component.ts

```diff
@@ -24,5 +65,166 @@ export class FindGardensComponent implements OnInit {
     nav: true,
     margin: 20,
   };
-  ngOnInit() {}
+
+  /**
+   * Initializes the component based on provided arguments
+   *
+   * @param route Contains arguments.
+   */
+  constructor(private route: ActivatedRoute, private httpClient: HttpClient) {}
+
+  ngOnInit(): void {
+    const zipCodeArg = this.route.snapshot.paramMap.get('zip-code');
+    this.createNearbyGardenList(zipCodeArg);
+  }
+
+  /**
+   * Finds gardens close to a given zip code.
+   * Returns an observable HTTP response.
+   *
+   * Performs GET: /find-gardens[?zip-code={zipCode}]
+   * If the argument zipCode is falsy, the query string
+   * is omitted.
+   *
+   * @param zipCode The zipcode to send to the server, or a falsy value to
+   * search near the current user's location.
+   * @return the http response.
+   */
+  getNearbyGardenList(
+    zipCode: string | null | undefined
```

Comment:
Why is it necessary to have `null` and `undefined`?

Reply:
`...paramMap.get()` returns `string | null`, so I will remove undefined and make appropriate comment changes